### PR TITLE
Add support for new GA failure domain labels

### PR DESCRIPTION
**Pull request by johscheuer**
Please provide a description for what this PR is for.

Currently Pilot supports for the locality based routing the well-known labels `failure-domain.beta.kubernetes.io/region` and `failure-domain.beta.kubernetes.io/zone`. Kubelet also supports the ga versions of this label: `failure-domain.kubernetes.io/zone` and `failure-domain.kubernetes.io/region` (as a fun fact kubelet also complains about the "old" labels). In our scenario/cluster we only set the new ga labels and would still like to benefit from the locality based routing. This PR adds the functionality for pilot to read the `zone` and `region` from the ga labels as well (actually as fallback if the old labels are not set).

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -47,10 +47,14 @@ import (
...
 
 const (
-	// NodeRegionLabel is the well-known label for kubernetes node region
+	// NodeRegionLabel is the well-known label for kubernetes node region in beta
 	NodeRegionLabel = "failure-domain.beta.kubernetes.io/region"
-	// NodeZoneLabel is the well-known label for kubernetes node zone
+	// NodeZoneLabel is the well-known label for kubernetes node zone in beta
 	NodeZoneLabel = "failure-domain.beta.kubernetes.io/zone"
+	// NodeRegionLabelGA is the well-known label for kubernetes node region in ga
+	NodeRegionLabelGA = "failure-domain.beta.kubernetes.io/region"
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/controller.go]: looks like the same label?

**Comment by johscheuer** [comment]
Are there actually any reasons that the controller code doesn't use the labels from: https://github.com/kubernetes/api/blob/master/core/v1/well_known_labels.go#L21 ?

**Comment by howardjohn** [comment]
And I think the reason we have the labels copied is that pulling just those in was pulling in tons of dependencies. That may not be the case anymore, kubernetes has done some work breaking out dependencies

**Pull request merged**